ui/pages/asset/components/asset-v2.tsx

```diff
@@ -0,0 +1,510 @@
+import React, { useContext, useEffect, useState, useRef } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useHistory } from 'react-router-dom';
+import { ReactNode } from 'react-markdown';
+import {
+  getCurrentChainId,
+  getCurrentCurrency,
+  getCurrentKeyring,
+  getIsBridgeChain,
+  getIsBuyableChain,
+  getIsSwapsChain,
+  getMetaMetricsId,
+  getSwapsDefaultToken,
+} from '../../../selectors';
+import {
+  BackgroundColor,
+  BlockSize,
+  Display,
+  FlexDirection,
+  JustifyContent,
+  TextColor,
+  TextVariant,
+} from '../../../helpers/constants/design-system';
+import {
+  Box,
+  ButtonPrimary,
+  ButtonSecondary,
+  ButtonSecondarySize,
+  Text,
+} from '../../../components/component-library';
+import { formatCurrency } from '../../../helpers/utils/confirm-tx.util';
+import { hexToDecimal } from '../../../../shared/modules/conversion.utils';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import {
+  AddressCopyButton,
+  TokenListItem,
+} from '../../../components/multichain';
+import {
+  MetaMetricsEventCategory,
+  MetaMetricsEventName,
+  MetaMetricsSwapsEventSource,
+} from '../../../../shared/constants/metametrics';
+import { MetaMetricsContext } from '../../../contexts/metametrics';
+import { startNewDraftTransaction } from '../../../ducks/send';
+import {
+  AssetType,
+  TokenStandard,
+} from '../../../../shared/constants/transaction';
+import {
+  BUILD_QUOTE_ROUTE,
+  SEND_ROUTE,
+} from '../../../helpers/constants/routes';
+import { INVALID_ASSET_TYPE } from '../../../helpers/constants/error-keys';
+import fetchWithCache from '../../../../shared/lib/fetch-with-cache';
+import { MINUTE } from '../../../../shared/constants/time';
+import TokenCell from '../../../components/app/token-cell';
+import TransactionList from '../../../components/app/transaction-list';
+import Tooltip from '../../../components/ui/tooltip';
+import { getPortfolioUrl } from '../../../helpers/utils/portfolio';
+import useRamps from '../../../hooks/experiences/useRamps';
+import { setSwapsFromToken } from '../../../ducks/swaps/swaps';
+import { isHardwareKeyring } from '../../../helpers/utils/hardware';
+import AssetChart from './asset-chart';
+import { getPricePrecision, localizeLargeNumber } from './util';
+import AssetHeader from './asset-header';
+
+const renderRow = (leftColumn: string, rightColumn: ReactNode) => (
+  <Box display={Display.Flex} justifyContent={JustifyContent.spaceBetween}>
+    <Text color={TextColor.textAlternative} variant={TextVariant.bodyMdMedium}>
+      {leftColumn}
+    </Text>
+    {rightColumn}
+  </Box>
+);
+
+const renderTooltip = (button: JSX.Element, text: string, show: boolean) =>
+  show ? (
+    <Tooltip wrapperClassName="asset-tooltip" title={text} position="bottom">
+      {button}
+    </Tooltip>
+  ) : (
+    <Box width={BlockSize.Full}>{button}</Box>
+  );
+
+// A page representing details for a native or token asset
+const AssetV2 = ({
```

Comment:
emm wondering if it's better to split this one on several components because of the size

Reply:
I split out the buy/swap/send/bridge buttons